presto-benchmark/src/main/java/com/facebook/presto/benchmark/SqlRewriteConditionalAggregationBenchmarks.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.benchmark;
+
+import com.facebook.presto.testing.LocalQueryRunner;
+import com.google.common.collect.ImmutableMap;
+
+import static com.facebook.presto.benchmark.BenchmarkQueryRunner.createLocalQueryRunner;
+
+public class SqlRewriteConditionalAggregationBenchmarks
+        extends AbstractSqlBenchmark
+{
+    public SqlRewriteConditionalAggregationBenchmarks(LocalQueryRunner localQueryRunner, String sql)
+    {
+        super(localQueryRunner, "sql_rewrite_conditional_aggregation", 5, 10, sql);
+    }
+
+    public static void main(String[] args)
+    {
+        String sql = "SELECT IF(GROUPING(suppkey, orderkey, partkey) = from_base('000', 2), sum(quantity)),"
+                + " IF(GROUPING(suppkey, orderkey, partkey) = from_base('001', 2), count_if(quantity>10)),"
+                + " IF(GROUPING(suppkey, orderkey, partkey) = from_base('010', 2), min(quantity)),"
+                + " IF(GROUPING(suppkey, orderkey, partkey) = from_base('011', 2), max(quantity)),"
+                + " IF(GROUPING(suppkey, orderkey, partkey) = from_base('100', 2), sum(extendedprice)),"
+                + " IF(GROUPING(suppkey, orderkey, partkey) = from_base('101', 2), count_if(extendedprice>100)),"
+                + " IF(GROUPING(suppkey, orderkey, partkey) = from_base('110', 2), min(extendedprice)),"
+                + " IF(GROUPING(suppkey, orderkey, partkey) = from_base('111', 2), max(extendedprice))"
+                + " FROM lineitem GROUP BY CUBE (suppkey, orderkey, partkey)";
+        System.out.println("Without optimization");
```

Comment:
Let's use logger instead of doing raw stdout

Reply:
Done